tests/foreman/api/test_permission.py

```diff
@@ -41,17 +63,17 @@ def test_search_by_name(self, permission_name):
         is the one searched for.
 
         """
+        permission_name = random.choice(self.permission_names)
         if (
                 permission_name == 'power_compute_resources_vms' and
                 bz_bug_is_open(1198731)):
-            self.skipTest('BZ 1198731 is open.')
+            permission_name = random.choice(self.permission_names)
```

Comment:
Oh, I get it. This test just checks to see whether it's possible to search by name, and `test_search_permissions` checks to see if permissions have been added or removed. That makes sense. Lacking the ability to use `subTest` or something similar, your current approach might make the most sense.